aws-rds-integration/aws-rds-integration.json

```diff
@@ -0,0 +1,166 @@
+      "items": {
+        "$ref": "#/definitions/Tag"
+      }
+    },
+    "SourceArn": {
+      "type": "string",
+      "description": "The Amazon Resource Name (ARN) of the Aurora DB cluster to use as the source for replication."
+    },
+    "TargetArn": {
+      "type": "string",
+      "description": "The ARN of the Redshift data warehouse to use as the target for replication."
+    },
+    "IntegrationArn": {
+      "type": "string",
+      "description": "The ARN of the integration."
+    },
+    "KMSKeyId": {
+      "type": "string",
+      "description": "The AWS Key Management System (AWS KMS) key identifier for the key used to to encrypt the integration. "
+    },
+    "AdditionalEncryptionContext": {
+      "type": "array",
+      "insertionOrder": false,
+      "items": {
+        "$ref": "#/definitions/EncryptionContextNameValuePair"
+      },
+      "description": "An optional set of non-secret key\u2013value pairs that contains additional contextual information about the data."
+    },
+    "CreateTime": {
+      "type": "string"
+    }
+  },
+  "required": [
+    "SourceArn",
+    "TargetArn",
+    "IntegrationName"
+  ],
+  "definitions": {
+    "Tags": {
+      "type": "array",
+      "maxItems": 50,
+      "uniqueItems": true,
+      "insertionOrder": false,
+      "description": "An array of key-value pairs to apply to this resource.",
+      "items": {
+        "$ref": "#/definitions/Tag"
+      }
+    },
+    "Tag": {
+      "description": "A key-value pair to associate with a resource.",
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "Key": {
+          "type": "string",
+          "description": "The key name of the tag. You can specify a value that is 1 to 128 Unicode characters in length and cannot be prefixed with aws:. You can use any of the following characters: the set of Unicode letters, digits, whitespace, _, ., /, =, +, and -. ",
+          "minLength": 1,
+          "maxLength": 128
+        },
+        "Value": {
+          "type": "string",
+          "description": "The value for the tag. You can specify a value that is 0 to 256 Unicode characters in length and cannot be prefixed with aws:. You can use any of the following characters: the set of Unicode letters, digits, whitespace, _, ., /, =, +, and -. ",
+          "minLength": 0,
+          "maxLength": 256
+        }
+      },
+      "required": [
+        "Key"
+      ]
+    },
+    "EncryptionContextNameValuePair": {
+      "description": "A name-value pair to associate with a resource.",
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "Name": {
+          "type": "string",
+          "description": "TODO",
+          "minLength": 1
+        },
+        "Value": {
+          "type": "string",
+          "description": "TODO",
+          "minLength": 0
+        }
+      },
+      "required": [
+        "Name"
+      ]
+    }
+  },
+  "propertyTransform": {
```

Comment:
I have improved the documentation - It is whatever KMS's DescribeKey API accepts ([link](https://docs.aws.amazon.com/kms/latest/APIReference/API_DescribeKey.html#API_DescribeKey_RequestSyntax)). I see that the API documentation is very vague on this as well, I will get in touch with our doc writer to address this.

RE: alias, I did not understand your concern in that I am not sure what kind of transformation it would accept, since the RDS API accepts all forms already.

Reply:
If the customer provided a KMS Key id, but the describe call returns the key ARN, this will cause false drift. For example, in DBInstance, the customer can use the KMS key ID, but the describe call returns the full KMS key ARN. CloudFormation will think this is drift, while both of them are the same thing. We use "propertyTransform" as a guard against it. Here is an example: [DBInstance Schema KMS Key property transform](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-rds/blob/master/aws-rds-dbinstance/aws-rds-dbinstance.json#L480)

Unfortunately, if a customer is using a key alias, they will see false drift as well, as CloudFormation has not yet implemented a solution for it.

Reply:
@moataz-mhmd It looks like the integration API will _always_ normalize it to the full key ARN, so customers shouldn't really use the alias for this. I'll update the schema & document to reflect that.

I was thinking technically we _can_ solve the issue for the customers, if the read handler can invoke kms:DescribeKey and decide which ARN form to produce based on the current resource model. Maybe an item for future improvement.

Reply:
Cool, 
For future improvement idea, regrettably, it will not work as the read handler does not have access to the resource model when the detect drift operation is triggered.